quiltctl/command/init/prompter.js

```
@@ -0,0 +1,428 @@
+/* eslint no-console: 0 */
+
+const path = require('path');
+const os = require('os');
+const fs = require('fs');
+
+const inquirer = require('inquirer');
+const consts = require('./constants');
+const Provider = require('./provider');
+const util = require('../../../util/util');
+
+
+/**
+  * Return a list of all provider names as listed in providers.json.
+  *
+  * @return {string[]} A list of provider names.
+  */
+function allProviders() {
+  const providerFile = path.join(__dirname, 'providers.json');
+  const providerInfo = JSON.parse(fs.readFileSync(providerFile, 'utf8'));
+  return Object.keys(providerInfo);
+}
+/**
+  * Check if the Quilt SSH keys exists.
+  *
+  * @return {boolean} True iff both the private and public Quilt SSH keys exist.
+  */
+function quiltSshKeyExists() {
+  return (fs.existsSync(consts.quiltSshKeyLocationPublic) &&
+      fs.existsSync(consts.quiltSshKeyLocationPrivate));
+}
+
+/**
+  * Returns true if the input string is a number. Otherwise return a string.
+  *
+  * @param {string} input The string to check.
+  * @return {boolean|string} True if the input string is a number. Otherwise
```

This is a little weird.  I wonder if it would be cleaner to throw an exception rather than rely on the return value?